### ISSUE TITLE
Make better Error notifications when usingen SomeOf()

### ISSUE THREAD

**Issue by kevinkit** [aleju/imgaug#300]
I met the Error "AssertionError: Assertion Failed" and I could not really find out what went wrong, due to the nature of `SomeOf`.

Here are some exemplary Codes that will fail:

```
seq=[      
 iaa.SomeOf((0,2)),[
      iaa.AdditiveGaussianNoise(loc=0,scale=(0.0,0.05*255),per_channel=0.5),
      iaa.Add((-10,10),per_channel=0.5),
      iaa.AddToHueAndSaturation((-20,20))]]


seq = iaa.Sequential(seq)
```

In my opinion it is hard to see why there is an error, cause the code can be exeucted correctly, till the last step. 

```
seq=[      
 iaa.SomeOf(2),[
      iaa.AdditiveGaussianNoise(loc=0,scale=(0.0,0.05*255),per_channel=0.5),
      iaa.Add((-10,10),per_channel=0.5),
      iaa.AddToHueAndSaturation((-20,20))]]

seq = iaa.Sequential(seq)
```

Here it is a little bit clearer what the problem is, the `SomeOf `Function was closed too early, hence the correct code should be:

```
seq=[      
 iaa.SomeOf((0,2),[
      iaa.AdditiveGaussianNoise(loc=0,scale=(0.0,0.05*255),per_channel=0.5),
      iaa.Add((-10,10),per_channel=0.5),
      iaa.AddToHueAndSaturation((-20,20))])]

seq = iaa.Sequential(seq)
```

Is there the possibility to have a better error mesage for `SomeOf`,something like "Missing augmenter?" which would indicate, that the someOf Function is not used correctly, instead of an AssertionError?

**Comment by aleju** [comment]
Yes, there are quite some places in the code where there are no proper error messages. That can then be a bit difficult to debug, though the stacktrace should usually give an indication about where it failed.

In these cases it did not fail in `SomeOf`, as that augmenter can handle getting no child augmenters. It fails at `iaa.Sequential(seq)` as that augmenter verifies that all entries in the list that it gets are `Augmenter` instances, but in your failing examples one of them is again a list.

Btw, in your specific example you can skip the `seq = iaa.Sequential(seq)` case as `SomeOf` already contains all augmenters that you are using.

**Comment by aleju** [comment]
PR #387 added descriptive error messages to all `assert` statement in the library.
